learning_observer/learning_observer/auth/social_sso.py

```diff
@@ -129,6 +141,10 @@ async def social_handler(request):
         )
 
     user = await _google(request)
+    
+    roster_source = await determine_roster_source(request)
```

Comment:
This is bad code. It should just be: 

```
roster_source = settings.pmss_settings.source(types=['roster_data'])
```

Good code here is not like an onion with layers. If it's this simple, it should be in-line.

Reply:
Removed the function